### PR TITLE
python3.pkgs.aiomysql: fix build dependency constraints

### PR DESCRIPTION
**Pull request by tjni**
## Description of changes

1. Pull patch that updates setuptools-scm and removes setuptools-scm-git-archive.
2. Add wheel as an explicit build dependency.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Shouldn't we wait until the PR you linked in the diff lands in a release?

**Comment by tjni** [comment]
It's unclear when a release will be made, and I need something like that for the work in https://github.com/NixOS/nixpkgs/issues/247287, because building a package with [`build`](https://pypa-build.readthedocs.io/en/latest/) validates build dependency constraints. At the moment, the failing constraint is that setuptools-scm in nixpkgs is too recent.

**Review by matthiasbeyer — APPROVED** [review]

**Comment by matthiasbeyer** [comment]
Result of `nixpkgs-review pr 250206` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.orm</li>
    <li>python310Packages.orm.dist</li>
    <li>python310Packages.ormar</li>
    <li>python310Packages.ormar.dist</li>
    <li>python311Packages.orm</li>
    <li>python311Packages.orm.dist</li>
    <li>python311Packages.ormar</li>
    <li>python311Packages.ormar.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.aiomysql</li>
    <li>python310Packages.aiomysql.dist</li>
    <li>python311Packages.aiomysql</li>
    <li>python311Packages.aiomysql.dist</li>
  </ul>
</details>

**Pull request merged**
